AbletonMCP_Remote_Script/handlers/devices.py

```
@@ -0,0 +1,172 @@
+        }
+    except Exception as e:
+        if ctrl:
+            ctrl.log_message("Error getting macro values: " + str(e))
+        raise
+
+
+def set_macro_value(song, track_index, device_index, macro_index, value, ctrl=None):
+    """Set the value of a specific macro control on a rack device."""
+    try:
+        if track_index < 0 or track_index >= len(song.tracks):
+            raise IndexError("Track index out of range")
```

Ruff (0.14.14)

[warning] Abstract `raise` to an inner function

(TRY301)